packages/eslint-config-nimble/typescript.js

```diff
@@ -0,0 +1,163 @@
+const restrictedImportsPaths = () => [
+    {
+        name: '@microsoft/fast-foundation',
+        importNames: ['focusVisible'],
+        message:
+            'Please use focusVisible from src/utilities/style/focus instead.'
+    },
+    {
+        name: '@microsoft/fast-components',
+        message:
+            'It is not expected to leverage @microsoft/fast-components directly as they are coupled to FAST design tokens.'
+    }
+];
+
+module.exports = {
+    plugins: ['jsdoc'],
+    extends: [
+        '@ni/eslint-config-typescript',
+        '@ni/eslint-config-typescript/requiring-type-checking'
+    ],
+    rules: {
+        // Require non-empty JSDoc comment on class declarations
+        'jsdoc/require-jsdoc': [
+            'error',
+            {
+                publicOnly: false,
+                require: {
+                    ClassDeclaration: true,
+                    FunctionDeclaration: false
+                }
+            }
+        ],
+        'jsdoc/require-description': [
+            'error',
+            { contexts: ['ClassDeclaration'] }
+        ],
+
+        'no-restricted-syntax': [
+            'error',
+            {
+                selector: 'TSEnumDeclaration',
+                message:
+                    'Use a const object instead of an enum. See other types.ts files for examples.'
+            }
+        ],
+
+        // Rules enabled due to strictNullChecks
+        '@typescript-eslint/no-non-null-assertion': 'off',
+
+        // Improves readability of templates to avoid return types in template expressions
+        '@typescript-eslint/explicit-function-return-type': [
+            'error',
+            { allowExpressions: true }
+        ],
+
+        'no-restricted-imports': [
+            'error',
+            {
+                paths: restrictedImportsPaths()
+            }
+        ],
+
+        // Enabled to prevent accidental usage of async-await
+        '@typescript-eslint/require-await': 'error'
+    },
+    ignorePatterns: ['.eslintrc.js'],
```

Comment:
huh, noticing it now but realize it was that way before. seems like the ignore patterns from the js should be here too, i.e. node_modules and dist:

```
    ignorePatterns: [
        '.eslintrc.js,
        'node_modules',
        'dist'
    ],
```

I don't actually know why `.eslintrc.js` is being ignored here, it can be formatted following our style rules too right?

Reply:
The reason we didn't ignore `node_modules` and `dist` is because this config was used within the `src` directory, so neither of those directories would be present. But since this config no longer assumes a context of the `src` directory, we can add those ignores. I'm also removing the ignore for `.eslintrc.js`. I don't know why we would need to exclude it.

Reply:
Turns out we were excluding `.eslintrc.js` for a reason. Without it, we get the following lint error:
> /home/runner/work/nimble/nimble/packages/nimble-components/src/.eslintrc.js
  0:0  error  Parsing error: ESLint was configured to run on `<tsconfigRootDir>/.eslintrc.js` using `parserOptions.project`: /home/runner/work/nimble/nimble/packages/nimble-components/tsconfig.json
However, that TSConfig does not include this file. Either:
> - Change ESLint's list of included files to not include this file
> - Change that TSConfig to include this file
> - Create a new TSConfig that includes this file and include it in your parserOptions.project
>
>See the typescript-eslint docs for more info: https://typescript-eslint.io/linting/troubleshooting#i-get-errors-telling-me-eslint-was-configured-to-run--however-that-tsconfig-does-not--none-of-those-tsconfigs-include-this-file

I tried adding `".eslintrc.js"` to the `"include"` of `nimble-components/tsconfig.json` (as that seems like what the second suggestion is saying), but that didn't work. I'm just going back to having the `.eslintrc.js` exclude itself.